common/namespace/cache.go

```diff
@@ -399,15 +401,28 @@ func (c *namespaceCache) GetNamespaceName(
 	return entry.info.Name, nil
 }
 
+func (c *namespaceCache) Refresh() {
```

Comment:
This is fine but I wonder if you considered returning `replyCh` so that the caller could decide if it wanted to block on completion (and for how long) or not.

Reply:
Can you think of any other potential callers than integration tests? If not, I'd rather just keep it simple.

I'm also assuming you're going to rewrite all this soon enough anyway, and maybe refresh won't even be needed.

Reply:
Fair enough. And yeah, I'm going to be mucking about in here a fair amount.